iroh/examples/search.rs

```diff
@@ -127,7 +127,8 @@ impl ProtocolHandler for BlobSearch {
     ///
     /// The returned future runs on a newly spawned tokio task, so it can run as long as
     /// the connection lasts.
-    fn accept(self: Arc<Self>, connecting: Connecting) -> BoxedFuture<Result<()>> {
+    fn accept(&self, connecting: Connecting) -> BoxedFuture<Result<()>> {
+        let this = self.clone();
```

Comment:
This clone is not needed by the router at all and only helpful for the implementations themselves, right?  So this is entirely an application-level concern.  But I guess it is a common pattern?

I wonder how to document this nicely so folks aren't lost searching for this for hours.